packages/store/src/types.js

```diff
@@ -482,143 +482,213 @@
- * Any elements beyond the `optional` pattern are matched against the
- * `rest` pattern.
+ * Matches any array --- typically an arguments list --- consisting of
+ *   - an initial portion matched by required, and
+ *   - a middle portion of length up to the length of optional that is
+ *     matched by the equal-length prefix of optional if optional is
+ *     defined, and
+ *   - a remainder that is matched by rest if rest is defined.
+ * The array must be at least as long as the required pattern,
+ * but its remainder can be arbitrarily short.
+ * Any array elements beyond the summed length of required and optional
+ * are collected and matched against the rest pattern.
+ *
  * @property {(required: CopyRecord<Pattern>,
  *             optional?: CopyRecord<Pattern>,
  *             rest?: Pattern,
  * ) => Matcher} splitRecord
- * Splits a copyRecord into those properties that match the `required`
- * pattern, the remainder which is matched against the `optional` pattern.
- * Any properties not matched by either are gathered into a record that
- * is matched against the `rest` pattern.
- * The specimen must have all the properties that appear on the `required`
- * pattern. It may omit properties that appear on the optional pattern.
- * For these purposes, `undefined` is like omission --- such properties
- * pass unconditionally.
+ * Matches any CopyRecord that can be split into component CopyRecords
+ * as follows:
+ *   - all properties corresponding with a property of required
+ *   - all properties corresponding with a property of optional
+ *     but not corresponding with a property of required
+ *   - all other properties
+ * where the first component is matched by the required pattern,
+ * the second component is matched by the subset of the optional pattern
+ * corresponding with its properties if optional is defined, and
+ * the third component is matched by the rest pattern if defined.
+ * The CopyRecord must have all properties that appear on the required
+ * pattern, but may omit properties that appear on the optional pattern.
  *
  * @property {(required: CopyRecord<*> | CopyArray<*>,
  *             rest?: Pattern,
  * ) => Matcher} split
  * Deprecated. Use `M.splitArray` or `M.splitRecord` instead.
- *
  * An array or record is split into the first part that matches the
  * base pattern, and the remainder, which matches against the optional
  * rest pattern if present.
+ *
  * @property {(base: CopyRecord<*> | CopyArray<*>,
  *             rest?: Pattern,
  * ) => Matcher} partial
  * Deprecated. Use `M.splitArray` or `M.splitRecord` instead.
- *
  * An array or record is split into the first part that matches the
  * base pattern, and the remainder, which matches against the optional
  * rest pattern if present.
  * Unlike `M.split`, `M.partial` ignores properties on the base
- * pattern that are not present on the specimen.
+ * pattern that are not present on the CopyRecord.
```

Comment:
Also, need to say the following somehow. I leave it to you to figure out how to work this in.

Unlike `M.split`, `M.partial` allows the CopyArray specimen to be shorter than the base pattern, and ignores all elements of the base pattern beyond those length limits.